From f0672a1e157c6acc50062dd28534a3e42e1b0f3f Mon Sep 17 00:00:00 2001
From: "Zhou, Furong" <furong.zhou@intel.com>
Date: Sat, 8 Mar 2025 19:11:18 +0800
Subject: [PATCH 74/76] virtio: fix VIRTIO_PMD support

The local variable disable_cb be referenced only if VIRTIO_PMD is
enableed. This will cause a 'unused variable' warning (or error
if Werror=unused-variable option enabled) if VIRETIO_PMD disabled.
---
 drivers/virtio/virtio_ring.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/virtio/virtio_ring.c b/drivers/virtio/virtio_ring.c
index bffb79e3c113f..62ebe6aa5a0f5 100644
--- a/drivers/virtio/virtio_ring.c
+++ b/drivers/virtio/virtio_ring.c
@@ -2629,7 +2629,9 @@ static struct virtqueue *__vring_new_virtqueue(unsigned int index,
 {
 	struct vring_virtqueue *vq;
 	int err;
+#ifdef CONFIG_VIRTIO_PMD
 	bool disable_cb = (callback == NULL);
+#endif
 
 	if (virtio_has_feature(vdev, VIRTIO_F_RING_PACKED))
 		return NULL;
-- 
2.43.0

